packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/eventbridge-scheduler/create-schedule.ts

```diff
@@ -0,0 +1,260 @@
+    this.taskPolicies = [
+      new iam.PolicyStatement({
+        resources: [
+          Stack.of(this).formatArn({
+            service: 'scheduler',
+            resource: 'schedule',
+            resourceName: `${this.props.groupName ?? 'default'}/${this.props.scheduleName}`,
+          }),
+        ],
+        actions: [
+          'scheduler:CreateSchedule',
+        ],
+      }),
+      new iam.PolicyStatement({
+        resources: [props.role.roleArn],
+        actions: [
+          'iam:PassRole',
+        ],
+      }),
+    ];
+  }
+
+  /**
+   * @internal
+   */
+  protected _renderTask(): any {
+    return {
+      Resource: integrationResourceArn('aws-sdk:scheduler', 'createSchedule', this.integrationPattern),
+      Parameters: {
+        ActionAfterCompletion: this.props.actionAfterCompletion,
+        ClientToken: this.props.clientToken,
+        Description: this.props.description,
+        EndDate: this.props.endDate ? this.props.endDate.toISOString() : undefined,
+        FlexibleTimeWindow: {
+          Mode: this.props.flexibleTimeWindow ? 'FLEXIBLE' : 'OFF',
+          MaximumWindowInMinutes: this.props.flexibleTimeWindow?.toMinutes(),
+        },
+        GroupName: this.props.groupName,
+        KmsKeyArn: this.props.kmsKey?.keyArn,
+        Name: this.props.scheduleName,
+        ScheduleExpression: this.props.scheduleExpression,
+        ScheduleExpressionTimezone: this.props.timezone,
+        StartDate: this.props.startDate ? this.props.startDate.toISOString() : undefined,
+        State: (this.props.enabled ?? true) ? 'ENABLED' : 'DISABLED',
+        Target: {
+          Arn: this.props.target,
+          RoleArn: this.props.role.roleArn,
+          Input: this.props.input,
+          RetryPolicy: this.props.retryPolicy ? {
+            MaximumEventAgeInSeconds: this.props.retryPolicy.maximumEventAge.toSeconds(),
+            MaximumRetryAttempts: this.props.retryPolicy.maximumRetryAttempts,
+          } : undefined,
+          DeadLetterConfig: this.props.deadLetterQueue ? {
+            Arn: this.props.deadLetterQueue.queueArn,
+          } : undefined,
+        },
+      },
+    };
+  }
+
+  private validateProps(props: EventBridgeSchedulerCreateScheduleTaskProps) {
+    if (
+      props.clientToken !== undefined &&
+      !Token.isUnresolved(props.clientToken) &&
+      (props.clientToken.length > 64 || props.clientToken.length < 1)) {
+      throw new Error(`ClientToken must be between 1 and 64 characters long. Got: ${props.clientToken.length}`);
+    }
+
+    if (props.description !== undefined && !Token.isUnresolved(props.description) && props.description.length > 512) {
+      throw new Error(`Description must be less than 512 characters long. Got: ${props.description.length}`);
+    }
+
+    if (props.flexibleTimeWindow && (props.flexibleTimeWindow.toMinutes() < 1 || props.flexibleTimeWindow.toMinutes() > 1440)) {
```

Comment:
We should check for `flexibleTimeWindow`, if specified and greater than 0, to be at least 1 minute to avoid unhandled exceptions.

Reply:
Sorry, I've understood.

I've updated validation condition like below.

```ts
    if (props.flexibleTimeWindow && (
      // To handle durations of less than one minute, comparisons will be made in milliseconds
      props.flexibleTimeWindow.toMilliseconds() < 60000 ||
      props.flexibleTimeWindow.toMinutes() > 1440
    )) {
      throw new Error('FlexibleTimeWindow must be between 1 and 1440 minutes');
    }
```